web/client/epics/annotations.js

```diff
@@ -436,15 +438,16 @@ export default (viewer) => ({
                 })
             );
         }),
-    showHideAnnotationEpic: (action$, store) => action$.ofType(TOGGLE_ANNOTATION_VISIBILITY)
+    showHideAnnotationEpic: (action$, store) => action$.ofType(TOGGLE_ANNOTATION_VISIBILITY, CHANGE_LAYER_PROPERTIES)
         .switchMap((action) => {
-            return Rx.Observable.of(
-                updateNode('annotations', 'layer', {
-                    features: annotationsLayerSelector(store.getState()).features.map(f => f.properties.id === action.id ? assign({}, f, {
-                        properties: {...f.properties, visibility: !isUndefined(f.properties.visibility) ? !f.properties.visibility : false}
-                    }) : f)
-                })
-            );
+            const feature = (f, visibility = false) => assign({}, f, {
+                properties: {...f.properties, visibility}
+            });
+            let isLayerPropertyChange = action.layer === "annotations";
+            const features = annotationsLayerSelector(store.getState()).features.map(f => isLayerPropertyChange ? feature(f, action?.newProperties?.visibility)
```

Comment:
This epic reacts on every `CHANGE_LAYER_PROPERTIES`. If annotation layer is not present, there is an error. Please prevent this.